Assets/Scripts/BoatMovement/BoatMovement.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Cinemachine;
+using TMPro;
+using UnityEngine.UI;
+
+public class BoatMovement : MonoBehaviour
+{
+    public float BoatSpeedValue;
+
+    [SerializeField] private float turnSpeed;
+    [SerializeField] private float boatSpeed;
+    [SerializeField] private Image speedometerArrow;
+
+    [SerializeField] private TextMeshProUGUI boatSpeedText;

```

Comment:
The movement script shouldn't have the responsibility of updating the UI. Consider creating a separate script that can reference this script and calculate the speed from there.